routes/api.php

```diff
@@ -34,17 +34,55 @@
-    Route::get('/events/{eventId}/dress-code','App\Http\Controllers\EventsController@fetchEventDresscode');
- 
+    Route::post('/createEvent','App\Http\Controllers\Event\EventsController@createEvent');
+    Route::get('/events/id={customerId}','App\Http\Controllers\Event\EventsController@fetchEvents');
+    Route::put('/events/id={eventId}','App\Http\Controllers\Event\EventsController@updateEvents');
+    Route::get('/fetchAllevents','App\Http\Controllers\Event\EventsController@fetchAllEvents');
+    Route::post('/publishEvent/id={eventId}','App\Http\Controllers\Event\EventsController@actionPublishEvent');
+    
+    // delete event api
+    Route::delete('/deleteAnEvent/id={eventId}','App\Http\Controllers\Event\EventsController@deleteAnEvent');
+
+    //fetch events
+    Route::get('/fetchAllActiveEvents','App\Http\Controllers\Event\EventsController@fetchAllActiveEvent');
+    Route::get('/fetchActiveEvents/id={customerId}','App\Http\Controllers\Event\EventsController@fetchActiveEvent');
+    Route::get('/fetchUpcomingEvents','App\Http\Controllers\Event\EventsController@fetchUpcomingEvent');
+    Route::get('/fetchCompletedEvent/id={customerId}','App\Http\Controllers\Event\EventsController@fetchCompletedEvent');
+    Route::get('/fetchUnpublishEvent/id={customerId}','App\Http\Controllers\Event\EventsController@fetchUnpublishEvent');
+    
+    
+    //event ticket api
+    Route::post('/events/{eventId}/tickets','App\Http\Controllers\Event\EventDashboard\TicketController@createEventTicket');
+    Route::get('/events/{eventId}/tickets','App\Http\Controllers\Event\EventDashboard\TicketController@fetchEventTicket');
+    Route::put('/events/{ticketId}/tickets','App\Http\Controllers\Event\EventDashboard\TicketController@updateEventTicket');
+    
+    //event dreescode api
+    Route::post('/events/{eventId}/dress-code','App\Http\Controllers\Event\EventDashboard\DressCodeController@createEventDresscode');
+    Route::get('/events/{eventId}/dress-code','App\Http\Controllers\Event\EventDashboard\DressCodeController@fetchEventDresscode');
+    
+    //event wishlist api
+    Route::get('/events/{eventId}/wishlist','App\Http\Controllers\Event\EventDashboard\WishlistController@fetchEventWishlist');
+    Route::post('/events/{eventId}/wishlist','App\Http\Controllers\Event\EventDashboard\WishlistController@createEventWishList');
+    Route::put('/events/{wishlistId}/wishlist','App\Http\Controllers\Event\EventDashboard\WishlistController@updateEventWishlist');
     //end events api 
+
+
+    //explore events
+    Route::get('/exploreCustomerEvents/id={customerId}','App\Http\Controllers\Explore\ExploreEventController@fetchExploreCustomerEvents');
+    Route::get('/exploreNearbyEvents','App\Http\Controllers\Explore\ExploreEventController@fetchExploreNearbyEvents');
+    Route::get('/fetchExploreEventDetails/id={eventId}','App\Http\Controllers\Explore\ExploreEventController@fetchExploreEventDetails');
```

Comment:
I think `events/{eventId}` should be fine here, I don't think we need to specify if it is from the explore page.

Reply:
reason behind using these naming convention is to better differentiate between modules(i.e explore , events, vendor etc)...so that it is easy for any developer  to find out the line/route in which he need to make changes.

Reply:
But your api clients (mobile guys etc) integrating your apis shouldn't care how the endpoint is structured on your server 🙂